charts/druid/templates/controller-deployment.yaml

```diff
@@ -2,18 +2,19 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .Release.Name }}
+  name: etcd-druid
   namespace: {{ .Release.Namespace }}
   labels:
+    app.kubernetes.io/name: etcd-druid
 spec:
   replicas: {{ .Values.replicas }}
   selector:
     matchLabels:
-      gardener.cloud/role: etcd-druid
+      app.kubernetes.io/name: etcd-druid
```

Comment:
I attempted to upgrade `etcd-druid` using the changes proposed in PR #777. After checking out the `master` branch and running `make deploy` in the local kind setup, I switched to the PR branch to deploy and encountered the following error during the upgrade process:

```bash
Error: UPGRADE FAILED: cannot patch "etcd-druid" with kind Deployment: Deployment.apps "etcd-druid" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/name":"etcd-druid"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
deploying "etcd-druid": install: exit status 1
m
```

Reply:
Nice find! @shreyas-s-rao i think the chart got changed and label-selector was changed. This cannot be done without deleting the deployment and recreating it.

Reply:
Addressed in https://github.com/gardener/etcd-druid/pull/777/commits/c1ea02b2acb462963f277d0e53a1534a68493f21